src/main/java/com/amplitude/Amplitude.java

```diff
@@ -44,18 +32,12 @@ public void init(String key) {
         apiKey = key;
     }
 
-    public void logEvent(String name) {
-        logEventWithProps(name, null);
+    public void logEvent(String name, String userId) {
+        logEvent(new Event(name, userId));
     }
 
-    public void logEventWithProps(String eventName, JSONObject eventProps) {
-        long time = System.currentTimeMillis();
-        Event event = new Event(eventName);
-        event.eventProperties = eventProps;
-        event.deviceId = deviceId;
-        event.userId = userId;
-        lastEventId++;
-        logEvent(event);
+    public void logEvent(String name, String userId, String deviceId) {
```

Comment:
One question here. In Node.js, userId and deviceId are semi-required.  That should be either they provide userId or deviceId or have all of those is also ok. 
In this case,  did we handle that customers only have deviceId no userId?

Reply:
They should use `logEvent(name, null, deviceId)`

Reply:
I think they might not need this func at all. Let's just keep `logEvent(Event e)`